cmd/src/debug_comp.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"archive/zip"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/sourcegraph/src-cli/internal/cmderrors"
+)
+
+func init() {
+	usage := `
+'src debug comp' mocks docker cli commands to gather information about a docker-compose Sourcegraph instance.
+
+Usage:
+
+    src debug comp [command options]
+
+Flags:
+
+	-o			Specify the name of the output zip archive.
+	-cfg		Include Sourcegraph configuration json. Defaults to true.
+
+Examples:
+
+    $ src debug comp -o debug.zip
+
+	$ src -v debug comp -cfg=false -o foo.zip
+
+`
+
+	flagSet := flag.NewFlagSet("comp", flag.ExitOnError)
+	var base string
+	var configs bool
+	flagSet.BoolVar(&configs, "cfg", true, "If true include Sourcegraph configuration files. Default value true.")
+	flagSet.StringVar(&base, "o", "debug.zip", "The name of the output zip archive")
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		//validate out flag
+		if base == "" {
+			return fmt.Errorf("empty -o flag")
+		}
+		// declare basedir for archive file structure
+		var baseDir string
+		if !strings.HasSuffix(base, ".zip") {
+			baseDir = base
+			base = base + ".zip"
+		} else {
+			baseDir = strings.TrimSuffix(base, ".zip")
+		}
+
+		// init context
+		ctx := context.Background()
+		// open pipe to output file
+		out, err := os.OpenFile(base, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0666)
+		if err != nil {
+			fmt.Errorf("failed to open file: %w", err)
```

Comment:
Error not handled here (and a few other places in this file)

Reply:
✅